src/main/java/org/sonarsource/sonarlint/ls/connected/notifications/ServerNotifications.java

```diff
@@ -0,0 +1,182 @@
+      coreNotifications().remove(config.listener());
+    }
+  }
+
+  private void registerConfigurationIfNeeded(@Nullable String connectionId, @Nullable String projectKey) {
+    if (!alreadyHasConfiguration(connectionId, projectKey)) {
+      if(!connections.containsKey(connectionId) || connections.get(connectionId).isDevNotificationsDisabled()) {
+        // Connection is unknown, or has notifications disabled - do nothing
+        return;
+      }
+      logMessage(String.format("Enabling notifications for %s on %s", projectKey, connectionId));
+      NotificationConfiguration newConfiguration = newNotificationConfiguration(connections.get(connectionId), projectKey);
+      coreNotifications().register(newConfiguration);
+      configurationsByProjectKeyByConnectionId.computeIfAbsent(connectionId, k -> new HashMap<>()).put(projectKey, newConfiguration);
+    }
+  }
+
+  private boolean alreadyHasConfiguration(@Nullable String connectionId, @Nullable String projectKey) {
+    return configurationsByProjectKeyByConnectionId.containsKey(connectionId) && configurationsByProjectKeyByConnectionId.get(connectionId).containsKey(projectKey);
+  }
+
+  private static org.sonarsource.sonarlint.core.notifications.ServerNotifications coreNotifications() {
+    return org.sonarsource.sonarlint.core.notifications.ServerNotifications.get();
+  }
+
+  private NotificationConfiguration newNotificationConfiguration(ServerConnectionSettings serverConnectionSettings, String projectKey) {
+    return new NotificationConfiguration(
+      new EventListener(serverConnectionSettings.isSonarCloudAlias()),
+      new ConnectionNotificationTime(),
+      projectKey,
+      () -> projectBindingManager.createServerConfiguration(serverConnectionSettings.getServerId()));
+  }
+
+  private void logMessage(String message) {
+    logOutput.log(message, LogOutput.Level.DEBUG);
+  }
+
+  /**
+   * Simply displays the events and discards it
+   */
+  private class EventListener implements ServerNotificationListener {
+
+    private final boolean isSonarCloud;
+
+    EventListener(boolean isSonarCloud) {
+      this.isSonarCloud = isSonarCloud;
+    }
+
+    @Override
+    public void handle(ServerNotification serverNotification) {
+      final String category = serverNotification.category();
+      telemetry.devNotificationsReceived(category);
+      final String label = isSonarCloud ? "SonarCloud" : "SonarQube";
+      ShowMessageRequestParams params = new ShowMessageRequestParams();
+      params.setType(MessageType.Info);
+      params.setMessage(String.format("%s Notification: %s", label, serverNotification.message()));
+      MessageActionItem browseAction = new MessageActionItem();
+      browseAction.setTitle("Open in " + label);
+      params.setActions(Collections.singletonList(browseAction));
+      client.showMessageRequest(params).thenAccept(action -> {
+        if(browseAction.equals(action)) {
+          client.browseTo(serverNotification.link());
```

Comment:
Here you should probably call the telemetry to report that the link has been clicked.